### ISSUE TITLE
Cannot configure with recent Intel compilers

### ISSUE THREAD

**Issue by prj-** [hypre-space/hypre#755]
It appears `configure` has been generated with a rather old `autoreconf` binary: https://github.com/hypre-space/hypre/blob/3c04212d77385869b7aa9a05fb07099c65a81b1b/src/configure#L4
It is causing grief with recent Intel compilers: https://gitlab.com/petsc/petsc/-/issues/1255#note_1135549645. Could `configure` but updated somehow and regenerated with a newer binary? Thanks.

**Comment by rfalgout** [comment]
Hi @prj- .   There was too much going on in the petsc issue you pointed to above for me to see what's happening.  I would like to understand this before we try to update to a newer version of autoconf.  Could you provide a configure line with `config.log` etc. showing the problem?  Thanks!

**Comment by prj-** [comment]
Sure, here goes (could probably be trimmed some more but I already removed a good amount of the PETSc-generated stuff).
```
$ export I_MPI_CC=icx   
$ export I_MPI_CXX=icpx
$ export I_MPI_F90=ifort
$ ./configure CC="mpiicc" CFLAGS="-fPIC -O3 -fopenmp" CXX="mpiicpc" CXXFLAGS="-O3 -std=gnu++17 -fPIC -fopenmp" F90="mpiifort" F90FLAGS="-fPIC -g -O3 -fopenmp" FFLAGS="-fPIC -g -O3 -fopenmp" FC="mpiifort" F77="mpiifort" FCFLAGS="-fPIC -g -O3 -fopenmp" --enable-shared --with-MPI-lib-dirs="" --with-MPI-libs=""
[...] // runs to completion, OK
$ export I_MPI_F90=ifx
$ ./configure CC="mpiicc" CFLAGS="-fPIC -O3 -fopenmp" CXX="mpiicpc" CXXFLAGS="-O3 -std=gnu++17 -fPIC -fopenmp" F90="mpiifort" F90FLAGS="-fPIC -g -O3 -fopenmp" FFLAGS="-fPIC -g -O3 -fopenmp" FC="mpiifort" F77="mpiifort" FCFLAGS="-fPIC -g -O3 -fopenmp" --enable-shared --with-MPI-lib-dirs="" --with-MPI-libs=""
[...]
checking for dummy main to link with Fortran libraries... unknown
configure: error: in `/local/jolivet/petsc/arch-intel-c-opt/externalpackages/git.hypre/src':
configure: error: linking to Fortran libraries from C fails
See `config.log' for more details
```
I'm attaching [config.log](https://github.com/hypre-space/hypre/files/9786254/config.log), you'll see a random reference to "loopopt" which none of us over at PETSc knows what this is (probably related to https://dl.acm.org/doi/abs/10.1145/3493229.3493301?). Intel people suggest updating `autoconf` to resolve this, see https://community.intel.com/t5/Intel-Fortran-Compiler/ifx-2021-1-beta04-HPC-Toolkit-build-error-with-loopopt/m-p/1235375#M153121.

**Comment by prj-** [comment]
Disclaimer: I have actually no idea if updating `configure` will fix anything, I'd be happy to try any potential fix for integration.

**Comment by rfalgout** [comment]
> Disclaimer: I have actually no idea if updating `configure` will fix anything, I'd be happy to try any potential fix for integration.

Okay, thanks.  This helps.  Try this branch: https://github.com/hypre-space/hypre/tree/autoconf-update.  Note that you can try other versions of autoconf available to you by just going into `hypre/src` and running `config/update.sh`.

**Comment by prj-** [comment]
Thanks, that does it (for `configure` and `make`) -- @ericch1, you can try the branch with `--download-hypre-commit=origin/autoconf-update`.

My bad, I didn't know about `config/update.sh`, I would have submitted the "fix" otherwise.

**Comment by rfalgout** [comment]
This changes `configure` rather substantially, so I will want to do some additional testing before making the switch.  Let me know how it goes on your end.  Thanks!

**Comment by prj-** [comment]
No problem, I'll launch a PETSc pipeline later today. Thank you.

**Comment by prj-** [comment]
It's breaking one of our FreeBSD build: https://gitlab.com/petsc/petsc/-/jobs/3176125524/artifacts/browse/arch-ci-freebsd-pkgs-opt/lib/petsc/conf/. @balay, do you know how to make sense out of:
```
Making seq_mv ...
gmake[1]: Entering directory '/home/svcpetsc/builds/RppTjuEb/0/petsc/petsc/arch-ci-freebsd-pkgs-opt/externalpackages/git.hypre/src/seq_mv'
make autogen
gmake[1]: Leaving directory '/home/svcpetsc/builds/RppTjuEb/0/petsc/petsc/arch-ci-freebsd-pkgs-opt/externalpackages/git.hypre/src/seq_mv'usage: make [-BeikNnqrstWwX] 
            [-C directory] [-D variable] [-d flags] [-f makefile]
            [-I directory] [-J private] [-j max_jobs] [-m directory] [-T file]
            [-V variable] [-v variable] [variable=value] [target ...]
gmake[1]: *** [Makefile:90: all] Error 2
gmake: *** [Makefile:86: all] Error 1
```

**Comment by balay** [comment]
> Executing: ./configure --prefix=/home/svcpetsc/petsc-hash-pkgs/6f7ef7 MAKE=/usr/local/bin/gmake 

> checking whether /usr/local/bin/gmake sets $(MAKE)... yes

> make autogen

The above  probably be `/usr/local/bin/gmake autogen`

**Comment by prj-** [comment]
Good catch.
https://github.com/hypre-space/hypre/blob/5546cc22d46b3dba253849f258786da47c9a7b21/src/seq_mv/Makefile#L90

**Comment by prj-** [comment]
Turns out this "bug" is also in `v2.16.0`, so we can't upgrade to this release in PETSc (at least on the FreeBSD boxes). Rob, the following patch fixes things: https://github.com/prj-/hypre/commit/2eee2dbbccab02555282677c324d6e818832f3bb. It's applied on top of your `autoconf-update` branch, but it should apply to `master` as well. With the branch from my fork, PETSc pipeline is clean on FreeBSD (https://gitlab.com/petsc/petsc/-/pipelines/668363360), and this also fixes the present issue (with Intel compilers).

**Comment by rfalgout** [comment]
Hi @prj- .  I made a few changes in the `autoconf-update` branch.  Let me know if this works without the patch.

Also, note that I still need to have some discussion with the hypre team about upgrading to autoconf 2.71 (and do more testing).  This may be a pain for us.  Hopefully not.  I'll get back to you on that.  Thanks!

**Comment by prj-** [comment]
New (partial) pipeline is here: https://gitlab.com/petsc/petsc/-/pipelines/669400607. Looks like it is clean, thank you for the hot fix. You probably want to decorrelate the fix for Intel compilers from the one for the wrongful use of `make` instead of `$(MAKE)`. Hopefully, the latter will be easy to integrate (just two different SLOC). For Intel compilers, we have about 6 months ahead of us for the next PETSc release, so I'm not in a hurry. Thanks.

**Comment by prj-** [comment]
I've opened a [PR](https://github.com/hypre-space/hypre/pull/763) for the `makefile` fix.

**Comment by victorapm** [comment]
Closing this issue since it has been resolved with #765
